packages/net-stubbing/lib/server/middleware/response.ts

```diff
@@ -65,7 +65,13 @@ export const InterceptResponse: ResponseMiddleware = async function () {
   }
 
   const mergeChanges = (before: CyHttpMessages.IncomingResponse, after: CyHttpMessages.IncomingResponse) => {
-    _.merge(before, _.pick(after, SERIALIZABLE_RES_PROPS))
+    _.mergeWith(before, _.pick(after, SERIALIZABLE_RES_PROPS), (_a, b) => {
+      if (b instanceof Buffer) {
+        return b
+      }
+
+      return undefined
+    })
```

Comment:
`_.merge` [converts Buffers to Arrays](https://github.com/lodash/lodash/issues/2964), so we use `_.mergeWith` with a customizer that preserves the source buffer.

Reply:
the same bug on the request-side was recently fixed like so: https://github.com/cypress-io/cypress/blob/9333bb61e5c88c7ce7ebca9b2351d63170d7325f/packages/net-stubbing/lib/server/middleware/request.ts#L145-L156

could you please update request.ts to use your implementation (which looks cleaner) and import it from somewhere so it can be shared b/t response.ts and request.ts?

Reply:
31ee892 suggestions for a better name are highly welcome :-P